anesthetic/samples.py

```diff
@@ -129,45 +129,51 @@ def plot(self, ax, paramname_x, paramname_y=None, *args, **kwargs):
 
         if do_1d_plot:
             if paramname_x in self and plot_type is not None:
-                x = self[paramname_x].compress()
                 xmin, xmax = self._limits(paramname_x)
                 if plot_type == 'kde':
-                    plot = plot_1d
+                    x = self[paramname_x].compress()
+                    return plot_1d(ax, x, xmin=xmin, xmax=xmax,
+                                   *args, **kwargs)
                 elif plot_type == 'hist':
-                    plot = hist_1d
+                    return hist_1d(ax, self[paramname_x], weights=self.w,
+                                   xmin=xmin, xmax=xmax, *args, **kwargs)
```

Comment:
It is the weights passed here that breaks the interface with astropy and that also prevents the `str` options for `bins` in numpy and matplotlib's histogram functions. I agree that it might be simpler to drop any interface with astropy, however, I find it a pity if none of the `str` options work anymore.

Could we turn the data into unweighted data before passing it to `hist_1d`?

Reply:
Weighted samples are going to be quite important for future development, so I'm keen to have hist_1d accepting weights. 

I have another PR almost ready for pushing that follows from this one, implementing scipy.stats.gaussian_kde as an alternative to fastkde. fastkde uses unweighted samples, whilst gaussian_kde uses weighted ones. 

A consistent way of reinstating astropy would be to have a separate 'astropyhist' option in addition to 'hist'. Would that do as a compromise?

Reply:
OK, Pushed the above. I think it's better anyway to explicitly state that you're using astropy, rather than letting the system quietly replace it. Resolves #47

Reply:
Is there a reason why weighted samples are important in the context of histograms? Otherwise, wouldn't it be easy enough to turn a weighted sample into an unweighted one just before passing it to `hist_1d`? That wouldn't interfere with weighted samples at all...

Reply:
Ah, well spotted. I had forgotten to put in that compression step before the astropyhist.

In general, it's always better to use the weights if you can. Compressing is not lossless, and it's better to let the underlying algorithms use the weights as best they see fit.

Reply:
Hmm, I guess you are thinking more in terms of nested sampling whereas I've been thinking more in terms of MCMC chains where the weights are integers. With integer weights there would be no problem turning weighted into unweighted samples, right? Or am I missing something?

Reply:
That's right. It's trivial for MCMC, but for nested sampling because the weights are continuous (and a large fraction are close to zero) making this compression is not trivial, and not lossless. The best you can do is provided by the compress functionality of WeightedDataFrame.

Note that even for MCMC, you could run into memory issues if your chain is already large, and you intend to increase its size by several factors by simply repeating samples.

Reply:
Hmm, ok. 

How accurate is the `compress()` function? Could this cause noticeable differences in the appearance of a histogram (for a sufficiently large sample size)?

Reply:
For large sample sizes the difference should  be negligible. Why?

Reply:
`compress()`, `channel_capacity()` etc. are not super intuitive in how they work. Just making sure that they don't have a visible effect on histograms that I wasn't aware of.

It's nice to have `'astropyhist'` as an option.